### PR TITLE
[Network] Network Watcher: Connection-Monitor and Show-Topology features

### PR DESCRIPTION
**Pull request by tjprescott**
Adds Connection-Monitor commands. Updates `show-topology` command to include new parameters. Breaking change is in `route-filter rule create`.  The service no longer supports adding tags to rules, so the parameter was removed.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [X] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [X] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/5742
This is an **experimental preview** for @microsoft users.

**Comment by tjprescott** [comment]
@lmazuel need a release of the latest network SDK.

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py [hunk 1]
```diff
@@ -865,11 +894,72 @@ def process_nw_flow_log_show_namespace(cmd, namespace):
...
+
+    if rg and not vnet and not subnet:
+        # targeting resource group - OK
+        pass
+    elif subnet:
+        subnet_usage = CLIError('usage error: --subnet ID | --subnet NAME --resource-group NAME --vnet NAME')
+        # targeting subnet - OK
+        if subnet_id and (vnet or rg):
+            raise subnet_usage
+        elif not subnet_id and (not rg or not vnet or vnet_id):
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py]: based on the error message, should we check the subnet , say `not subnet_id and (not rg or not vnet or not subnet or vnet_id)`?

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py [hunk 2]
```diff
@@ -865,11 +895,72 @@ def process_nw_flow_log_show_namespace(cmd, namespace):
...
+            )
+            namespace.target_resource_group_name = None
+            namespace.target_vnet = None
+            namespace.target_subnet = SubResource(subnet_id)
+    elif vnet:
+        # targeting vnet - OK
+        vnet_usage = CLIError('usage error: --vnet ID | --vnet NAME --resource-group NAME')
+        if vnet_id and (subnet or rg):
+            raise vnet_usage
+        elif not vnet_id and not rg or subnet:
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py]: Based on the error message, should it be `not vnet_id and (not rg or not vnet)`?

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py (hunk 1 above)
Comment (tjprescott) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py]: I'll test and check.

**Comment by tjprescott** [comment]
Three backup tests needed to be re-recorded due to GitHub rebase shenanigans, but something is odd with the test infrastructure and they will not record. Temporarily converted to LiveScenarioTest since they recorded fine yesterday.  Will TS after this is merged.

**Pull request merged**